tests/integration/aof.tcl

```diff
@@ -485,4 +485,30 @@ tags {"aof external:skip"} {
         catch {exec src/redis-check-aof --truncate-to-timestamp 1628217469 $aof_path} e
         assert_match {*aborting*} $e
     }
+
+    test {EVAL timeout with slow verbatim Lua script from AOF} {
+        create_aof {
+            append_to_aof [formatCommand select 9]
+            append_to_aof [formatCommand eval {redis.call('set',KEYS[1],'y'); for i=1,1500000 do redis.call('ping') end return 'ok'} 1 x]
+        }
+
+        start_server [list overrides [list dir $server_path appendonly no lua-time-limit 1 aof-use-rdb-preamble no]] {
+            # generate a long running script that is propagated to the AOF as script
+            # make sure that the script times out during loading
+            set rd [redis_deferring_client]
+            r config set appendonly yes
+            set start [clock clicks -milliseconds]
+            $rd debug loadaof
+            $rd flush
+            after 100
+            catch {r ping} err
+            assert_match {LOADING*} $err
+            $rd read
+            set elapsed [expr [clock clicks -milliseconds]-$start]
+            if {$::verbose} { puts "loading took $elapsed milliseconds" }
+            $rd close
+            assert_equal [r get x] y
```

Comment:
If the script is timed out then I'd expect an error not "y"

Reply:
'timeout' isn't equal to script kill. After $rd read, the script has finished its operations successfully.

Reply:
i think this test is ok (timeout means redis responds to command during the script, it didn't fail).
specifically, it's the same test we had before, but because of our change the old test needed adjustments, to still cover cases of loading old AOF files.
the test itself may not be perfect, and although we can work on it to improve it, maybe we can also leave it imperfect.